lib/ews/connection.rb

```diff
@@ -67,6 +68,14 @@ def authenticate
     self.get && true
   end
 
+  def set_logger(logger:)
+    @logger = logger
+  end
+
+  def set_user_agent(user_agent:)
```

Comment:
`def user_agent=(user_agent:)`? or maybe just change it to `attr_accessor :logger, :user_agent`?